docs/design.md

```diff
@@ -0,0 +1,59 @@
+# OSBA Design
+
+We can roughly split OSBA's architecture into three pieces:
+
+1.  REST API frontend
+2.  Stable Storage
+3.  Asynchronous processing
+
+The remainder of this document will overview each piece.
+
+(Note: testing infrastructure is out of scope for this document)
+
+# REST API Frontend
+
+OSBA runs an HTTP server that provides a REST API that conforms to the
+[Open Service Broker API specification](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md)
+
+TODOs for this section:
+
+* Specific version of the spec?
```

Comment:
@krancour @jeremyrickard are you targeting a specific version/range in the spec?

Reply:
We haven't tied to a specific version yet, just 2.x.